pkg/apis/core/types.go

```diff
@@ -1603,6 +1606,51 @@ type CSIPersistentVolumeSource struct {
 	NodePublishSecretRef *SecretReference
 }
 
+// Represents a source location of a volume to mount, managed by an external CSI driver
+type CSIVolumeSource struct {
+	// Driver is the name of the driver to use for this volume.
+	// Required.
+	Driver string
+
+	// VolumeHandle is the unique volume name returned by the CSI volume
+	// plugin’s CreateVolume to refer to the volume on all subsequent calls.
+	// If not provided, that handle will be auto-generated.
+	// +optional
+	VolumeHandle *string
+
+	// Optional: The value to pass to ControllerPublishVolumeRequest.
+	// Defaults to false (read/write).
+	// +optional
+	ReadOnly *bool
+
+	// Filesystem type to mount. Ex. "ext4", "xfs", "ntfs".
+	// If not provided, the empty value is passed to the associated CSI driver
+	// which will determine the default filesystem to apply.
+	// +optional
+	FSType *string
+
+	// VolumeAttributes store immutable properties of the volume copied during provision.
+	// These attributes are passed back to the driver during controller publish calls.
+	// +optional
+	VolumeAttributes map[string]string
```

Comment:
will this include the file permission bits ?

Reply:
This is an interesting question. Today, secrets have a setting allows the user to [set the mode](https://sourcegraph.com/github.com/kubernetes/kubernetes/-/blob/pkg/apis/core/types.go#L853), but it is not well integrated with the volume ownership (fsgroup) feature.  For alpha, I think drivers can have an attribute for this, but we may need to revisit adding first class support for a better experience.

Reply:
I think in the future (beta) we should continue to use  attributes or some other semi-structured parameter map to avoid making CSI ephemeral feature looking like any existing ephemeral driver.  It should be more flexible to support unforeseen use cases.

Reply:
Flex volumes were used as ephemeral until now and I haven't heard anyone complaining that permissions were missing there.